graveler/sstable/writer.go

```diff
@@ -122,6 +123,10 @@ func (dw *DiskWriter) Abort() error {
 }
 
 func (dw *DiskWriter) Close() (*committed.WriteResult, error) {
+	// Before closing, we write all user supplied metadata keys and values to the hash
+	// This is done to avoid collisions, especially on empty sstables that might hash to the same value otherwise.
+	ident.WriterFromHash(dw.hash).MarshalStringMap(dw.props)
```

Comment:
I don't understand why this drags `ident` in like this.  We already have `dw.hash`, it is a `Writer`.  If `ident.MarshalStringMap` were a function then I think you could just say
```suggestion
	ident.MarshalStringMap(dw.hash, dw.props)
```
I think that would make it easier for me to understand the line.

Reply:
I agree, refactored ident to make this possible. PTAL